add godesim & gitaligned

soypat opened the pull request:
Proud to present my first open source project, `godesim`! 
**Disclaimer**: This is the second attempt at adding this library. I have now matched the requirements for a data library to be added.

- github.com repo: https://github.com/soypat/godesim
- pkg.go.dev: https://pkg.go.dev/github.com/soypat/godesim
- goreportcard.com: https://goreportcard.com/report/github.com/soypat/godesim
- coverage service link [codecov](https://codecov.io/gh/soypat/godesim/branch/main)
[![Go Report Card](https://goreportcard.com/badge/github.com/soypat/godesim)](https://goreportcard.com/report/github.com/soypat/godesim)
[![go.dev reference](https://pkg.go.dev/badge/github.com/soypat/godesim)](https://pkg.go.dev/github.com/soypat/godesim)
[![codecov](https://codecov.io/gh/soypat/godesim/branch/main/graph/badge.svg)](https://codecov.io/gh/soypat/godesim/branch/main)
---

- [x] I have added my package in alphabetical order.
- [x] I have an appropriate description with correct grammar.
- [x] I know that this package was not listed before.
- [x] I have added pkg.go.dev link to the repo and to my pull request.
- [x] I have added coverage service link to the repo and to my pull request.
- [x] I have added goreportcard link to the repo and to my pull request.
- [x] I have read [Contribution guidelines](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standard).

avelino commented:
Thank you for contributing with [awesome-go](https://github.com/avelino/awesome-go), we will revise your contribution as soon as possible.

Automation body links content check:
- godoc.org or pkg.go.dev: True
- goreportcard.com: True
- coverage: True

> your project is about to be approved, it's under revision, _it may take a few days_

soypat commented:
I wasn't sure how to go about adding another project while my first one was pending: here's `gitaligned`, a project that surprised me with the amount of positive feedback it received.

- github.com repo: https://github.com/soypat/gitaligned
- pkg.go.dev: https://pkg.go.dev/github.com/soypat/gitaligned
- goreportcard.com: https://goreportcard.com/report/github.com/soypat/gitaligned
- coverage service link [codecov](https://codecov.io/gh/soypat/gitaligned/branch/main)
[![Go Report Card](https://goreportcard.com/badge/github.com/soypat/gitaligned)](https://goreportcard.com/report/github.com/soypat/gitaligned)
[![go.dev reference](https://pkg.go.dev/badge/github.com/soypat/gitaligned)](https://pkg.go.dev/github.com/soypat/gitaligned)
[![codecov](https://codecov.io/gh/soypat/gitaligned/branch/main/graph/badge.svg)](https://codecov.io/gh/soypat/gitaligned/branch/main)
---

- [x] I have added my package in alphabetical order.
- [x] I have an appropriate description with correct grammar.
- [x] I know that this package was not listed before.
- [x] I have added pkg.go.dev link to the repo and to my pull request.
- [x] I have added coverage service link to the repo and to my pull request.
- [x] I have added goreportcard link to the repo and to my pull request.
- [x] I have read [Contribution guidelines](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standard).

avelino commented on README.md:
```diff
@@ -1579,6 +1580,7 @@ See [go-hardware](https://github.com/rakyll/go-hardware) for a comprehensive lis
 * [go-dsp](https://github.com/mjibson/go-dsp) - Digital Signal Processing for Go.
 * [go-estimate](https://github.com/milosgajdos/go-estimate) - State estimation and filtering algorithms in Go.
 * [go-gt](https://github.com/ThePaw/go-gt) - Graph theory algorithms written in "Go" language.
+* [godesim](https://github.com/soypat/godesim) - Extended/multivariable ODE solver framework for event-based simulations with simple API.
```
one package per contribution, divide this PR in 2

soypat replied:
@avelino `gitaligned` removed. PR should contain only `godesim`.
